ledger/blockqueue.go

```diff
@@ -164,6 +164,22 @@ func (bq *blockQueue) syncer() {
 			bq.mu.Unlock()
 
 			minToSave := bq.l.notifyCommit(committed)
+			var earliest basics.Round
+			const maxDeletionBatchSize = 10_000
```

Comment:
Does `sqlite` support and `LIMIT` in `DELETE` statements.  A nice easy fix would be to add a `LIMIT` to:
```
	_, err = tx.Exec("DELETE FROM blocks WHERE rnd<?", rnd)
```
in `BlockForgetBefore`.

Reply:
It does https://www.sqlitetutorial.net/sqlite-delete/

We might also need to add an `ORDER BY rnd` (I could not tell whether ORDER BY was required when using LIMIT.

Reply:
It is compiled with it [enabled](https://github.com/mattn/go-sqlite3/blob/00b02e0ba98effd5f157d39216e244af8a807f9b/sqlite3.go#L22) so I guess we could. Not sure if there performance penalty, looks like the same as query in advance + a plain delete 

```
EXPLAIN QUERY PLAN DELETE FROM blocks WHERE rnd<1999 ORDER BY rnd LIMIT 10000;
QUERY PLAN
|--SEARCH blocks USING INTEGER PRIMARY KEY (rowid=?)
`--LIST SUBQUERY 1
   `--SEARCH blocks USING INTEGER PRIMARY KEY (rowid<?)

EXPLAIN QUERY PLAN DELETE FROM blocks WHERE rnd<1999;
QUERY PLAN
`--SEARCH blocks USING INTEGER PRIMARY KEY (rowid<?)
vs
```

Reply:
I tried
```
_, err = tx.Exec("DELETE FROM blocks WHERE rnd<? ORDER BY rnd LIMIT 10000", rnd)
```
and it fails with `near \"ORDER\": syntax error" file=blockqueue.go`. Looks like it is not enabled in the driver for whatever reason.

Reply:
Can you trying skipping the "ORDER BY rnd" part?  If that doesn't work, I guess it's not worth investigating. (I don't think the ORDER BY should really be needed.)

Reply:
Sure. `"near \"LIMIT\": syntax error"`

Reply:
oh well, it was a beautiful dream